### ISSUE TITLE
Failed to convert 'mime_content' into class <str>

### ISSUE THREAD

**Issue by immez** [ecederstrand/exchangelib#683]
I have encountered this error several times: Cannot convert value '{some chars looks like base64encoded}' on field 'mime_content' to type <class 'str'>
When the error occurs, item.mime_content will be None, and sometimes the other properties of the mail are also failed to be parsed, e.g. item.datetime_received was set to be None.
What can I do for this error? 

Sorry for that I can't give the example mime_content here for privacy reasons.

**Comment by immez** [comment]
sometimes the mime_content will be None, but the other properties will be ok, and sometimes, seems like all properties were not parsed

**Comment by immez** [comment]
I raised the error at line 542 of file fields.py, it says that binascii.a2b_base64 appeared 'Incorrect padding'.
I changed line 540 into return base64.b64decode(val+'=='), and then the mime_content seems to be success converted, but there are still no other properties of the item.

**Comment by ecederstrand** [comment]
This is a duplicate of #670 

It's interesting that it works if you add the missing padding. This indicates that EWS is sending invalid base64-encoded data.

Can you confirm if the padding is also missing in the original response XML? Here's how to print the raw XML response: https://github.com/ecederstrand/exchangelib#troubleshooting

**Comment by ecederstrand** [comment]
Ah, it seems it's actually a bug in Python 3.7 and 3.8: https://bugs.python.org/issue33770

In short, the extra trailing padding in base64-encoded strings is optional.

I'll see if I can provide a workaround.

**Comment by immez** [comment]
There are so many info in the logging output
I am not sure which response is the correct one, so I output the logging into a file and then 
with open('mail.out', 'r+', encoding='utf8') as f:text = f.read()
for t in re.findall('<t:MimeContent.+?>(.+?)</t:MimeContent>', text):print(len(t)/4.0)
all the result are integers

but when i insert a "print(len(val))" after line 538(fields.py), the output is a number cannot be divided by 4. and there are no more output(the other mails will output a number 32)
--------------
and there are two points I noticed but have no idea about them:
1, i tryed to add 1 or 2 '=' in line 540, in both tests the mime_content part were ok, and it did not affect the other mails, (I thought that it should be make the correct mine_content to be odds?)
2, this issue appears more likely on linux, (12+ of 1200 mails). i tryed to process the same mails on windows, it appeared only once, and just the mime_content was None, but on linux, everything is None.

**Comment by ecederstrand** [comment]
I think the referenced patch should fix your issue with empty `mime_content`. Feel free to reopen if it didn't.

**Comment by immez** [comment]
Thanks, but I still encoutered this:
  File "/usr/local/lib/python3.6/site-packages/exchangelib/queryset.py", line 305, in __getitem__
    return self._getitem_idx(idx_or_slice)
  File "/usr/local/lib/python3.6/site-packages/exchangelib/queryset.py", line 321, in _getitem_idx
    for item in new_qs.__iter__():
  File "/usr/local/lib/python3.6/site-packages/exchangelib/queryset.py", line 290, in __iter__
    for val in self._format_items(items=self._query(), return_format=self.return_format):
  File "/usr/local/lib/python3.6/site-packages/exchangelib/queryset.py", line 367, in _item_yielder
    for i in iterable:
  File "/usr/local/lib/python3.6/site-packages/exchangelib/account.py", line 605, in fetch
    item = validation_folder.item_model_from_tag(i.tag).from_xml(elem=i, account=self)
  File "/usr/local/lib/python3.6/site-packages/exchangelib/items.py", line 169, in from_xml
    item = super(BaseItem, cls).from_xml(elem=elem, account=account)
  File "/usr/local/lib/python3.6/site-packages/exchangelib/properties.py", line 1234, in from_xml
    kwargs = {f.name: f.from_xml(elem=elem, account=account) for f in cls.supported_fields()}
  File "/usr/local/lib/python3.6/site-packages/exchangelib/properties.py", line 1234, in <dictcomp>
    kwargs = {f.name: f.from_xml(elem=elem, account=account) for f in cls.supported_fields()}
  File "/usr/local/lib/python3.6/site-packages/exchangelib/fields.py", line 559, in from_xml
    val = super(MimeContentField, self).from_xml(elem=elem, account=account)
  File "/usr/local/lib/python3.6/site-packages/exchangelib/fields.py", line 543, in from_xml
    raise e
  File "/usr/local/lib/python3.6/site-packages/exchangelib/fields.py", line 540, in from_xml
    return safe_b64decode(val)
  File "/usr/local/lib/python3.6/site-packages/exchangelib/util.py", line 296, in safe_b64decode
    data += b'=' * (4 - overflow)
TypeError: must be str, not bytes

I replaced b'=' wirh '=', this error disappeared, but there are still no datetime_received, no sender, no other values in the item. Does exchangelib extract these values from mime_content? Does exchangelib extract these values from mime_content before the mime_content was converted to str?

**Comment by ecederstrand** [comment]
Fixed your TypeError in the referenced commit.

We don't fetch other field values from mime_content, so that must be a separate issue. Are they mentioned in the response XML? Did you explicitly exclude them with `.only('mime_content')`?

**Comment by immez** [comment]
![image](https://user-images.githubusercontent.com/7035123/69813798-eac79c80-122d-11ea-8b00-7ba0d172b3f7.png)
Yes, i am sure that they were mentioned in the response XML, (Sorry for the shadow because of privacy reasons).

And I have no idea about “.only('mime_content')”. I just wanted to read all the mails, and most of them goes right, but some of them raised Error on mime_content. Now the mime_content error was fixed by safe_base64decode, but there are still no other properties.

I guess that there are some odd characters in the response XML, but I dont know how to examine it and cannot paste the whole XML for privacy policy.
